### PR TITLE
Split Configure API into Content, Model API paths

### PR DESCRIPTION
**Pull request by debanjum**
## Major: Breaking Changes
- Move API endpoints under /configure/<type>/model to /api/model/<type>
- Move API endpoints under /api/configure/content/ to /api/content/
- Accept file deletion requests by clients during sync
- Split /api/v1/index/update into /api/content PUT, PATCH API endpoints

## Minor: Create New API Endpoint
- Create API endpoints to get user content configurations

Related: #852

### CODE REVIEW

File: src/khoj/routers/api.py [hunk 1]
```diff
@@ -276,6 +278,49 @@ async def transcribe(
...
+    request: Request,
+    name: str,
+    client: Optional[str] = None,
+):
+    user = request.user.object
+
+    split_name = name.split(" ")
+
+    if len(split_name) > 2:
+        raise HTTPException(status_code=400, detail="Name must be in the format: Firstname Lastname")
```
Comment (sabaimran) [inline src/khoj/routers/api.py:304]: Is this rule necessary? I guess it's fine if people want to have their middle name's or something. We can just put a character limit.

File: src/khoj/routers/api_content.py [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+                image=None,
+                github=None,
+                notion=None,
+                plaintext=None,
+            )
+            state.config.content_type = default_content_config
+            save_config_to_file_updated_state()
+            configure_search(state.search_models, state.config.search_type)
+
+        # Extract required fields from config
```
Comment (sabaimran) [inline src/khoj/routers/api_content.py:395]: Is this log line relevant?

**Review by sabaimran — APPROVED** [review]
Neat getting the API to a state where it's more semantically coherent.

File: src/khoj/routers/api.py (hunk 1 above)
Comment (debanjum) [reply src/khoj/routers/api.py:304]: Not it's not actually required. We could just split into last word = last name and everthing else = first name and put a character limit. But maybe can deal with improving the username management separately?

File: src/khoj/routers/api_content.py (hunk 2 above)
Comment (debanjum) [reply src/khoj/routers/api_content.py]: Fixed

**Pull request merged**